scanner/src/main/kotlin/Scanner.kt

```diff
@@ -767,3 +779,10 @@ private fun ScanController.getSubRepositories(id: Identifier): Map<String, VcsIn
 
 private fun ScanResult.clearPackageVerificationCode(): ScanResult =
     copy(summary = summary.copy(packageVerificationCode = ""))
+
+private fun ProvenanceResolutionResult.filterByVcsPath(): ProvenanceResolutionResult =
+    copy(
+        subRepositories = subRepositories.filter { (path, _) ->
+            File(path).startsWith(File(packageProvenance?.vcsPath.orEmpty()))
```

Comment:
`File.startsWith()` has an overload that takes a `String` that could be used here.